### PR TITLE
feat: support dynamic provisioning

### PR DESCRIPTION
**Pull request by andyzhangx**
update doc

**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
feat: support dynamic provisioning

```yaml
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  name: smb
provisioner: smb.csi.k8s.io
parameters:
  source: "//smb-server.default.svc.cluster.local/share"
  csi.storage.k8s.io/node-stage-secret-name: "smbcreds"
  csi.storage.k8s.io/node-stage-secret-namespace: "default"
  createSubDir: "true"  # optional: create a sub dir for new volume
reclaimPolicy: Retain  # only retain is supported
volumeBindingMode: Immediate
mountOptions:
  - dir_mode=0777
  - file_mode=0777
  - uid=1001
  - gid=1001
```

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [x] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
feat: support dynamic provisioning
```

### CODE REVIEW

File: deploy/example/e2e_usage.md [hunk 1]
```diff
@@ -1,14 +1,57 @@
...
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: smb
+provisioner: smb.csi.k8s.io
+parameters:
+  source: "//smb-server.default.svc.cluster.local/share"
+  csi.storage.k8s.io/node-stage-secret-name: "smbcreds"
+  csi.storage.k8s.io/node-stage-secret-namespace: "default"
+  createSubDir: "true"  # optional: create a sub dir for new volume
```
Comment (msau42) [inline deploy/example/e2e_usage.md:24]: Is this actually supported yet? I don't see any implementation here: https://github.com/kubernetes-csi/csi-driver-smb/blob/master/pkg/smb/controllerserver.go#L29

Comment (andyzhangx) [reply deploy/example/e2e_usage.md:24]: yes, there is a  trick here, I created a sub dir in NodeStageVolume:
https://github.com/kubernetes-csi/csi-driver-smb/blob/84ac39ff780b8db59b29f7f64d6dda2154356c13/pkg/smb/nodeserver.go#L95-L100

Comment (msau42) [reply deploy/example/e2e_usage.md:24]: Then there is no way to clean it up when the pvc gets deleted?

Comment (andyzhangx) [reply deploy/example/e2e_usage.md:24]: yes, so `reclaimPolicy: Retain  # only retain is supported`

**Pull request merged**